site/guides/integration-testing.adoc

```diff
@@ -0,0 +1,970 @@
+====
+
+
+==== How FMP resources are used before the tests are run
+
+Now that we have a firm grasp on how and when FMP creates the necessary OpenShift resources, we will close the loop and describe how these resources are used
+in order to launch the application before Arquillian starts the actual tests.
+
+As should be evident from the previous section, by the time Arquillian launches (as part of the `integration-test` phase of the default Maven lifecycle),
+FMP has taken care of two things:
+
+* An `ImageStream` has been applied on OpenShift that contains the Docker image built from the application's uber-jar
+* A YAML file containing all the OpenShift resources necessary to launch the application has been created at `target/classes/META-INF/fabric8/openshift.yml`
+
+The aforementioned `openshift.yml` file has also been copied to the Spring Boot uber-jar (with the `META-INF` directory structure preserved).
+This is important since as was described earlier in the guide, the uber-jar is part of the test classpath created by the failsafe plugin.
+
+When Arquillian is launched, if it has been configured to have `env.init.enabled` set to `true`, it dutifully looks for `META-INF/fabric8/openshift.yml`
+inside it's classpath. Once the file is found, it used to deploy the application to OpenShift.
+
+The rest of what Arquillian does is the same as described in the section about testing an already running application.
+
+Just for reference, the Arquillian configuration we use in this case is:
+
+.src/test/resources/arquillian.xml
+----
+<arquillian xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns="http://jboss.org/schema/arquillian"
+            xsi:schemaLocation="http://jboss.org/schema/arquillian http://jboss.org/schema/arquillian/arquillian_1_0.xsd">
+
+  <extension qualifier="openshift">
+    <property name="namespace.use.current">true</property>
+    <property name="env.init.enabled">true</property>
+    <property name="enableImageStreamDetection">false</property>
+  </extension>
+
+</arquillian>
+----
+
+
+== Testing an application that uses a database
+
+In this section we will introduce some extra complexity into the integration testing scenario by using a
+link:https://github.com/snowdrop/spring-boot-crud-booster[demo application] that requires a database to be present.
+
+We will show how easy it is for Arquillian to take care of the deployment of a database as well as part of the test setup.
+
+=== Configuring Arquillian to additionally setup the database
+
+As has been noted before, Arquillian offers a ton of flexibility in order to be able to address a host of different
+integration testing scenarios. +
+In this section we will see just how a very simple configuration change to the configuration we have been using so far,
+suffices to get a database running on the OpenShift cluster.
+
+First of all we need to add an OpenShift file that describes the database we want to deploy:
+
+.src/test/resources/database.yml
```

Comment:
Maybe no need to show the file if we're not going to explain it? Might be interesting instead to explain how we could get such a file briefly (from service catalog for example?)…

Reply:
That is indeed a good point, but I don't really know where the proper place to get such a file would be. My guess is that it was just "exported" manually after the database was launched by the Service Broker.
We could explain that but I am afraid it would be too much. 
WDYT?